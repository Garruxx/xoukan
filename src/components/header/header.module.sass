.header
    display: flex
    width: 100%
    padding: var(--container-paddings)
    box-sizing: border-box
    overflow: hidden
    justify-content: space-between
    align-items: center
    background: var(--primary)
    img
        transition: .35s opacity
        cursor: pointer
        &:hover
            opacity: .7
        &:active
            opacity: .6
        
    div
        padding-right: 22px
        max-width: 700px
        h2
            font-size: var(--text-title)
            margin: 0 0 4px 0
        div
            background: var(--surface)
            padding: var(--container-paddings)
            p
                box-sizing: border-box
                overflow: auto
                height: 40px
                font-size: var(--text-medium)
                &::-webkit-scrollbar
                    width: 4px
                    background: var(--secondary)
                    border-radius: 4px
                &::-webkit-scrollbar-thumb
                    border-radius: 4px
                    background: var(--primary)

                span
                    display: block
                    margin-right: 4px
