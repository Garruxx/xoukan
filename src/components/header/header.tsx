import { HeaderProps } from './types/header.props'
import gitLogo from './assets/github-mark-white.svg'

export const Header = ({ conectionStatus, infoText }: HeaderProps) => {
	return (
		<div>
			<div>
				<h2>{conectionStatus}</h2>
				<div>
					<p>{infoText}</p>
				</div>
			</div>
			<img src={gitLogo} alt="git logo white svg png" />
		</div>
	)
}
