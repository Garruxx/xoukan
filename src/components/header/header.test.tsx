import { render } from '@testing-library/react'
import { Header } from './header'

describe('header', () => {
	test('should render', () => {
		const { baseElement } = render(
			<Header conectionStatus="" infoText="" />,
		)
		expect(baseElement).toBeDefined()
	})

	test('The connection status must be rendered', () => {
		const { getByText } = render(
			<Header conectionStatus="online" infoText="" />,
		)
		expect(getByText('online')).toBeDefined()
	})
	test('The infoText must be rendered', () => {
		const { getByText } = render(
			<Header conectionStatus="" infoText="infoText test" />,
		)
		expect(getByText('infoText test')).toBeDefined()
	})
})
