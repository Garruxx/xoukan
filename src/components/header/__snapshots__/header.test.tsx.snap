// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header Should renderer correctly 1`] = `"<div><div class="header"><div><h2>Online ⚡</h2><div><p><span>⚡ You are online ⚡ now only share and receive files</span></p></div></div><a href="https://github.com/Garruxx/xoukan" title="visit the repository" target="_blank"><img alt="git logo white svg png" width="40" height="40" src="[object Object]"></a></div></div>"`;
